src/shapes/object.class.js

```diff
@@ -1548,7 +1548,13 @@
 
       ctx.save();
       if (this.strokeUniform) {
-        ctx.scale(1 / this.scaleX, 1 / this.scaleY);
+        var scaleX = this.scaleX;
+        var scaleY = this.scaleY;
+        if (this.group) {
+          scaleX *= this.group.scaleX;
+          scaleY *= this.group.scaleY;
+        }
+        ctx.scale(1 / scaleX, 1 / scaleY);
```

Comment:
let's try to fix this.
This code will work only for object inside one group.
We need ALL the groups up to the canvas, and those can be more than 1.

Another example is that if you have an object rotated of 90degrees, the scaleX of the group will influence the scaleY of the object.

So things are never so easy.

If you want to bring this forward you can divide this code in 2 parts.

```js
if (this.strokeUniform && !this.group) {
// the old code, the most common case and more performant
} else if (this.strokeUniform) {
// there is a group then. Use calcTransformMatrix from the object, that will give you the full transform
// then decompose the matrix in scalex/scaleY operations.
var options = fabric.util.qrDecompose(this.calcTransformMatrix());
ctx.scale(1 / options.scaleX, 1 / options.scaleY);
}
```

See if this work, it would be a more generic case and would work with any rotation.